src/U1Manager.cpp

```
@@ -306,10 +306,109 @@ U1Manager& U1Manager::getInstance() {
+            strncpy(st.vendor, staged_.info.vendor, sizeof(st.vendor) - 1);
+            strncpy(st.material, staged_.info.main_type, sizeof(st.material) - 1);
+            st.rgb = staged_.info.rgb_1;
+        } else {
+            staged_.active = false;  // lazy expiry
+        }
+    }
+    taskEXIT_CRITICAL(&stagedMux_);
+    return st;
+}
+
+bool U1Manager::assignStagedToChannel(uint8_t channel) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)